runtime/compiler/optimizer/abstractinterpreter/AbsInterpreter.cpp

```diff
@@ -0,0 +1,3072 @@
+   /*** Case 3: Have more than one predecessors ***/
+   if (block->getPredecessors().size() > 1)
+      {
+      bool allPresecessorsInterpreted = true;
+
+      TR::Block *oneInterpretedBlock = NULL;
+
+      for (auto e = block->getPredecessors().begin(); e != block->getPredecessors().end(); e++)
+         {
+         TR::Block* predBlock = (*e)->getFrom()->asBlock();
+
+         if (getState(predBlock) == NULL) //any block is not interpreted
+            {
+            allPresecessorsInterpreted = false;
+            if (oneInterpretedBlock)
+               break;
+            }
+         else  //any block is interpreted
+            {
+            oneInterpretedBlock = predBlock;
+            }
+         }
+
+      /*** Case 3.1: All predecessors are interpreted. Merge their states ***/
+      if (allPresecessorsInterpreted)
+         {
+         TR::Block* firstPredBlock = (*block->getPredecessors().begin())->getFrom()->asBlock();
+         TR::AbsStackMachineState* state = getState(firstPredBlock)->clone(region());
+
+         for (auto e = ++block->getPredecessors().begin(); e != block->getPredecessors().end(); e ++)
+            {
+            TR::Block* predBlock = (*e)->getFrom()->asBlock();
+            state->merge(getState(predBlock), region());
+            }
+
+         setState(block, state);
+         return;
+         }
+
+      /*** Case 3.2: some blocks not interpreted ***/
+      if (insideLoop && !allPresecessorsInterpreted && oneInterpretedBlock)
+         {
+         TR::AbsStackMachineState* state = NULL;
+
+         for (auto e = ++block->getPredecessors().begin(); e != block->getPredecessors().end(); e ++)
+            {
+            TR::Block* predBlock = (*e)->getFrom()->asBlock();
+
+            if (!state && getState(predBlock))
+               state = getState(predBlock)->clone(region());
+            else if (getState(predBlock))
+               state->merge(getState(predBlock), region());
+            }
+
+         setState(block, state);
+         return;
+         }
+
+      /*** Case 3.3: set the state to TOP ***/
+      if (!allPresecessorsInterpreted && oneInterpretedBlock)
+         {
+         TR::AbsStackMachineState* copiedState = getState(oneInterpretedBlock)->clone(region());
+         for (size_t i = 0; i < copiedState->getArraySize(); i ++)
+            {
+            copiedState->at(i) ? copiedState->at(i)->setToTop() : copiedState->set(i, new (region()) TR::AbsVPValue(vp(), NULL, TR::NoType));
+            }
+
+         TR::vector<TR::AbsValue*, TR::Region&> temp(region());
+         while (copiedState->getStackSize() != 0)
+            {
+            TR::AbsValue* val = copiedState->pop();
+            if (val)
+               val->setToTop();
+            else
+               val = new (region()) TR::AbsVPValue(vp(), NULL, TR::NoType);
+
+            temp.push_back(val);
+            }
+
+         for (size_t i = 0; i < temp.size(); i ++)
+            {
+            copiedState->push(temp[i]);
```

Comment:
You are right, this is a weird implementation. I see the operands in `temp` are pushed in the stack in a FIFO way rather than a usual LIFO way to keep the operands in the final stack in the same order before the constraints are cleared out. However, this won't violate the assertion and we have already tested it.

I finally found the problem should be we made a same mistake when implemented the [`TR::AbsOpStack::clone()`](https://github.com/eclipse/omr/blob/0966eb360f6e0537e165426416051ac30c9afc65/compiler/optimizer/abstractinterpreter/AbsOpStack.cpp#L24-L32), which was called by `TR::AbsStackMachineState::clone()` and they would also give us a reversed operand stack. I'm not sure why the former contributor kept the mistake in `TR::AbsOpStack::clone()` rather than fixing it. 

Anyway, I will fix this in the next commit of the new phase-loop OMR PR.